### PR TITLE
test: download test fixtures differently

### PR DESCRIPTION
**Pull request by abrown**
The `openvino` crate uses some well-known models to smoke test that the bindings are working as expected. These models can be quite large (>100MB) and previously were embedded in the repository via Git LFS. This meant that, to retrieve them, one had to additionally run `git lfs fetch && git lfs checkout` or the tests would fail. This was both a bit unintuitive and resulted in Git LFS bandwidth charges from GitHub when the repository was retrieved frequently.

This change solves the problem by storing the test fixtures on https://download.01.org, downloading the fixtures on-demand using `curl`, and caching them if possible using the file system locally (they are not re-downloaded if present) and the GitHub actions cache in CI. Though this adds a dependency to run the tests, `curl`, I figure this dependency is more likely present on systems than `git-lfs`, so the trade seems a good one.

### CODE REVIEW

File: crates/openvino/tests/classify-alexnet.rs [hunk 1]
```diff
@@ -1,10 +1,10 @@
-//! Demonstrates using `openvino-rs` to classify an image using an AlexNet model and a prepared input tensor. See
-//! [README](fixtures/alexnet/README.md) for details on how this test fixture was prepared.
+//! Demonstrates using `openvino-rs` to classify an image using an AlexNet model and a prepared
+//! input tensor. See [`fixtures/alexnet`] for details on how this test fixture was prepared.
```
Comment (rahulchaphalkar) [inline crates/openvino/tests/classify-alexnet.rs:2]: `fixtures/alexnet` doesn't have details anymore, they're moved to `mod.rs` comments. Same for `classify-mobilenet.rs` and `classify-inception.rs`. I do see the Readme present in the remote location.

File: README.md [hunk 2]
```diff
@@ -20,26 +20,15 @@ crate (high-level, ergonomic bindings) for accessing OpenVINO™ functionality i
...
-   crates/openvino -name *.bin | xargs ls -lhr` returns `.bin` files of tens of megabytes. If not,
-   download the large files with:
-
-   ```shell
-   git lfs fetch
-   git lfs checkout
-   ```
-
-3. This library binds to OpenVINO™'s shared libraries; how those native libraries are configured and
+2. This library binds to OpenVINO™'s shared libraries; how those native libraries are configured and
```
Comment (rahulchaphalkar) [inline README.md:23]: NIT - Perhaps call out `curl` as a test dependency.

**Review by rahulchaphalkar — COMMENTED** [review]
Looks good! Definitely helps to get rid of `git-lfs` dep. A couple of Nits mentioned.

File: crates/openvino/tests/classify-alexnet.rs (hunk 1 above)
Comment (abrown) [reply crates/openvino/tests/classify-alexnet.rs]: I just decided to remove all of that; if anyone wants to know more, they should navigate to the `fixtures::alexnet` module and there's some documentation there they can use.

File: README.md (hunk 2 above)
Comment (abrown) [reply README.md:23]: Good call... added.

**Review by rahulchaphalkar — APPROVED** [review]
Looks good!

**Pull request merged**